blackjax/sgmcmc/csgld.py

```diff
@@ -0,0 +1,59 @@
+"""Public API for the Contour Stochastic gradient Langevin Dynamics kernel.
+.. [0]: Deng, W., Lin, G., Liang, F. (2020). 
+        A Contour Stochastic Gradient Langevin Dynamics Algorithm 
+        for Simulations of Multi-modal Distributions. 
+        In Neural Information Processing Systems (NeurIPS 2020).
+
+   [1]: Deng, W., Liang, S., Hao, B., Lin, G., Liang, F. (2022) 
+        Interacting Contour Stochastic Gradient Langevin Dynamics
+        In International Conference on Learning Representations (ICLR) 
+"""
+from typing import Callable, NamedTuple
+
+import jax
+import jax.numpy as jnp
+
+from blackjax.sgmcmc.diffusion import csgld
+from blackjax.types import PRNGKey, PyTree
+
+__all__ = ["CSGLDState", "init", "kernel"]
+
+
+class CSGLDState(NamedTuple):
+    step: int
+    position: PyTree
+    logprob_grad: PyTree
+    energy_pdf: PyTree
+    energy_idx: int
+
+
+def init(position: PyTree, 
+         batch, 
+         grad_estimator_fn: Callable, 
+         partition=512):
+    logprob_grad = grad_estimator_fn(position, batch)
+    energy_pdf = jnp.arange(partition, 0, -1) / jnp.arange(partition, 0, -1).sum()
+    return CSGLDState(0, position, logprob_grad, energy_pdf, partition-1)
+
+
+def kernel(grad_estimator_fn: Callable) -> Callable:
+    integrator = csgld(grad_estimator_fn, 
+                       zeta=1,
+                       temperature=0.01,
+                       partition=512,
+                       energy_gap=10,
+                       total_samples=60000)
```

Comment:
These parameters should be passed as arguments to `kernel` (besides `total_samples` which does not seem to be used).

Reply:
Not sure what the kernel means, let me check carefully and let you know later.

Reply:
do you mean i should define 
def kernel(grad_estimator_fn, zet, temperature, etc)?

Reply:
> These parameters should be passed as arguments to `kernel` (besides `total_samples` which does not seem to be used).

do you mean i should define
def kernel(grad_estimator_fn, zet, temperature, etc)?